app/models/tenant.rb

```diff
@@ -155,6 +155,18 @@ def login_logo?
     !!login_logo_file_name
   end
 
+  def visible_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).order('priority DESC')
+  end
+
+  def enabled_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id), :enabled => true).order('priority DESC')
+  end
+
+  def editable_domains
+    MiqAeDomain.where(:tenant_id => id, :system => false).order('priority DESC')
```

Comment:
We will have some overlap here and above for the work with Rbac we'll be doing to deal with searching for "things" (vms, domains, etc.) where we have to ask something (Tenant) what tenants are accessible from the current tenant.

I'll try to draft a branch with some of these questions/tests and possible designs today and tomorrow.